drivers/SmartThings/sonos/src/init.lua

```diff
@@ -73,17 +73,36 @@ local function _initialize_device(driver, device)
     end
 
     local fields = driver._field_cache[device.device_network_id]
-    driver._player_id_to_device[fields.player_id] = device -- quickly look up device from player id string
-    driver.sonos:mark_player_as_joined(fields.player_id)
 
     log.trace("Setting persistent fields")
     device:set_field(PlayerFields.WSS_URL, fields.wss_url, { persist = true })
     device:set_field(PlayerFields.HOUSEHOULD_ID, fields.household_id, { persist = true })
     device:set_field(PlayerFields.PLAYER_ID, fields.player_id, { persist = true })
 
-    device:set_field(PlayerFields._IS_INIT, true)
```

Comment:
Will this result in failure to rediscover a device whose ip address has changed since restarts?

Reply:
My instinct says that this should become eventually consistent as the group info comes in over the websocket subscriptions, but looking at this, there could definitely be edge cases where if the IP does change it may fail to make a websocket connection to a coordinator with visibility in to the speaker whose IP has changed. I'll have to think about this change hunk a bit more.